crates/bevy_core_pipeline/src/main_pass_3d.rs

```diff
@@ -55,7 +55,7 @@ impl Node for MainPass3dNode {
                 Err(_) => return Ok(()), // No window
             };
 
-        {
+        if !opaque_phase.items.is_empty() {
```

Comment:
Should we move this check inside the tracing span? This could impact profile output and averages across multiple frames if applied inconsistently.

Reply:
It may be a matter of expectations. 

If it's viewed from the perspective of "how much frame time does this section take on average?", I'd say counting the zeroes here is accurate, and properly reflects the performance changes that do affect the phase. If the phase is only in use 10% of the time, any optimization made within each will only impact 10% of the frames.

Reply:
I see your point. I just disagree. :)

Reply:
I’ll merge it like this and if it turns out to be the wrong call, it is easy to change. I’m thinking that frame time consistency is more important than throughout. If you do something once every 10 frames and it makes for a significant frame stutter, that’s bad. So averaging over only three instances where the thing actually happens seems like the preference. Maybe I’m wrong. Or maybe we need both for different situations.